### DISCUSSION TITLE
No `detektX` task is generated for additional source sets

### DISCUSSION THREAD

**Discussion by KotlinIsland** [detekt/detekt#6758] [Q&A] [answered]
```kt
plugins {
    kotlin("jvm") version "1.9.22"
    id("io.gitlab.arturbosch.detekt") version "1.23.4"
}

repositories {
    mavenCentral()
}

kotlin {
    sourceSets {
        create("other")
        main {
            dependsOn(getByName("other"))
        }
    }
}
```
No corresponding `detektOther`/`detektBaselineOther` is generated.

**Comment by BraisGabin** [comment]
Detekt doesn't work by source sets. Detekt works by compilations. So it doesn't have sense to have a task for a source set added to the main compilation.

That's the same reason we can not just inspect main on a multiplatform project. We need to inspect it once for each platform.

**Reply by KotlinIsland** [reply]
So, when I compile my code, the `other` source set isn't checked, despite being used. The problem is, I have another source set with source code in it, how do I check it?

**Reply by BraisGabin** [reply]
This seems like a tricky one. The generated folder is also compiled code that is inside the source set but no one wants to run detekt against those files. But I imagine that a manually added file should be added. CAN you create an issue pointing this problem? I don't know which would be the best solution here, it doesn't seem trivial.

Probably related with #6612

**Reply by KotlinIsland** [reply]
> CAN you create an issue pointing this problem?

I did, it was converted into this discussion.

**Reply by KotlinIsland** [reply]
https://github.com/detekt/detekt/issues/6796

**Comment by 3flex** [comment, marked as answer]
This feature is available on snapshot versions now but isn't documented yet as the design may change slightly: https://github.com/detekt/detekt/pull/6973
